st2common/bin/st2-run-pack-tests

```diff
@@ -205,9 +214,14 @@ if [ "${JUST_TESTS}" = false ]; then
     fi
 
     # Install global test dependencies
+    INSTALLED_PIP_PACKAGES=$(pip list)
+
     echo "Installing global pack test dependencies..."
-    for dependency in ${PACK_TEST_PYTHON_DEPENDENCIES[@]}; do
-        pip install --cache-dir ${HOME}/.pip-cache ${ST2_PIP_OPTIONS} "${dependency}"
+    for((i = 0; i < ${#PACK_TEST_PYTHON_DEPENDENCIES_NAMES[@]}; i++)); do
+        DEPENDENCY_NAME="${PACK_TEST_PYTHON_DEPENDENCIES_NAMES[$i]}"
+        DEPENDENCY_VERSION="${PACK_TEST_PYTHON_DEPENDENCIES_VERSIONS[$i]}"
+
+        echo ${INSTALLED_PIP_PACKAGES} | grep "${DEPENDENCY_NAME}"  > /dev/null || pip install --cache-dir ${HOME}/.pip-cache ${ST2_PIP_OPTIONS} "${DEPENDENCY_NAME}${DEPENDENCY_VERSION}"
```

Comment:
Yeah, not ideal, `pip install` is already idempotent if package is installed, but it still needs to read the local database, cache, etc. so actual pip invocation is slow as hell (https://github.com/StackStorm/st2/pull/3281#discussion_r105163117).